pkg/devicetwin/devicetwin_test.go

```diff
@@ -0,0 +1,193 @@
+	"github.com/kubeedge/kubeedge/beehive/pkg/core"
+	"github.com/kubeedge/kubeedge/beehive/pkg/core/context"
+	"github.com/kubeedge/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/mocks/beego"
+	"github.com/kubeedge/kubeedge/mocks/beehive"
+	"github.com/kubeedge/kubeedge/pkg/common/dbm"
+	"github.com/kubeedge/kubeedge/pkg/devicetwin/dtcommon"
+	"github.com/kubeedge/kubeedge/pkg/devicetwin/dtcontext"
+	"github.com/kubeedge/kubeedge/pkg/devicetwin/dtmodule"
+)
+
+const (
+	//TestModule is name of test.
+	TestModule = "test"
+	//GroupResource is name of resource group resource.
+	GroupResource = "resource"
+)
+
+// ormerMock is mocked Ormer implementation.
+var ormerMock *beego.MockOrmer
+
+// querySeterMock is mocked QuerySeter implementation.
+var querySeterMock *beego.MockQuerySeter
+
+// fakeModule is mocked implementation of TestModule.
+var fakeModule *beehive.MockModule
+
+// mainContext is beehive context used for communication between modules.
+var mainContext *context.Context
+
+// initMocks is function to initialize mocks.
+func initMocks(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	ormerMock = beego.NewMockOrmer(mockCtrl)
+	querySeterMock = beego.NewMockQuerySeter(mockCtrl)
+	fakeModule = beehive.NewMockModule(mockCtrl)
+	dbm.DBAccess = ormerMock
+}
+
+// registerFakeModule is fuction to register all mock modules used.
+func registerFakeModule() {
+	fakeModule.EXPECT().Name().Return(TestModule).Times(3)
+	core.Register(fakeModule)
+	mainContext = context.GetContext(context.MsgCtxTypeChannel)
+	mainContext.AddModule(TestModule)
+}
+
+// TestName is function to test Name().
+func TestName(t *testing.T) {
+	tests := []struct {
+		name string
+		want string
+	}{
+		{
+			name: "devicetwinnametest",
+			want: "twin",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dt := &DeviceTwin{}
+			if got := dt.Name(); got != tt.want {
+				t.Errorf("DeviceTwin.Name() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+// TestGroup is function to test Group().
+func TestGroup(t *testing.T) {
+	tests := []struct {
+		name string
+		want string
+	}{
+		{
+			name: "devicetwingrouptest",
+			want: "twin",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dt := &DeviceTwin{}
+			if got := dt.Group(); got != tt.want {
+				t.Errorf("DeviceTwin.Group() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+// TestStart is function to test Start().
+func TestStart(t *testing.T) {
+	initMocks(t)
+	registerFakeModule()
+	core.Register(&DeviceTwin{})
+	dt := DeviceTwin{}
+	mainContext.AddModule(dt.Name())
+	mainContext.AddModuleGroup(dt.Name(), dt.Group())
+	dt.context = mainContext
+	controller, err := InitDTController(mainContext)
+	if err != nil {
+		t.Errorf("Start device twin failed, due to %v", err)
+	}
+	dt.dtcontroller = controller
+	ormerMock.EXPECT().QueryTable(gomock.Any()).Return(querySeterMock).Times(200)
```

Comment:
Please give exact times this will be used. I think 200 is a random number correct?